File: Emulator/src/main/content/global/skill/summoning/familiar/npc/HydraNPC.java
package content.global.skill.summoning.familiar.npc;

import content.global.skill.farming.FarmingPatch;
import content.global.skill.farming.Patch;
import content.global.skill.summoning.familiar.Familiar;
import content.global.skill.summoning.familiar.FamiliarSpecial;
import core.game.node.Node;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.game.node.scenery.Scenery;
import core.plugin.Initializable;

/**
 * The type Hydra npc.
 */
@Initializable
public class HydraNPC extends content.global.skill.summoning.familiar.Familiar {

    /**
     * Instantiates a new Hydra npc.
     */
    public HydraNPC() {
        this(null, 6811);
    }

    /**
     * Instantiates a new Hydra npc.
     *
     * @param owner the owner
     * @param id    the id
     */
    public HydraNPC(Player owner, int id) {
        super(owner, id, 4900, 12025, 6, WeaponInterface.STYLE_RANGE_ACCURATE);
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new HydraNPC(owner, id);
    }

    @Override
    protected boolean specialMove(FamiliarSpecial special) {
        Node node = special.getNode();
        if (node instanceof Scenery) {
            Scenery scenery = (Scenery) node;
            FarmingPatch farmingPatch = FarmingPatch.forObject(scenery);
            if (farmingPatch != null) {
                Patch patch = farmingPatch.getPatchFor(owner, true);
                patch.regrowIfTreeStump();
                return true;
            }
        }

        return false;
    }

    @Override
    public int[] getIds() {
        return new int[]{6811, 6812};
    }

}

File: Emulator/src/main/content/global/skill/summoning/familiar/npc/HydraNPC.kt
package content.global.skill.summoning.familiar.npc

import content.global.skill.farming.FarmingPatch
import content.global.skill.summoning.familiar.Familiar
import content.global.skill.summoning.familiar.FamiliarSpecial
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.game.node.scenery.Scenery
import core.plugin.Initializable
import org.rs.consts.Items
import org.rs.consts.NPCs

/**
 * The type Hydra npc.
 */
@Initializable
class HydraNPC
/**
 * Instantiates a new Hydra npc.
 *
 * @param owner the owner
 * @param id    the id
 */
/**
 * Instantiates a new Hydra npc.
 */
@JvmOverloads constructor(owner: Player? = null, id: Int = NPCs.HYDRA_6811) :
    Familiar(owner, id, 4900, Items.HYDRA_POUCH_12025, 6, WeaponInterface.STYLE_RANGE_ACCURATE) {
    override fun construct(owner: Player, id: Int): Familiar {
        return HydraNPC(owner, id)
    }

    override fun specialMove(special: FamiliarSpecial): Boolean {
        val node = special.node
        if (node is Scenery) {
            val farmingPatch = FarmingPatch.forObject(node)
            if (farmingPatch != null) {
                val patch = farmingPatch.getPatchFor(owner, true)
                patch.regrowIfTreeStump()
                return true
            }
        }

        return false
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.HYDRA_6811, NPCs.HYDRA_6812)
    }
}
